### PR TITLE
test(bob): refactor integration tests

### PR DESCRIPTION
**Pull request by smol-ninja**
Partially addresses https://github.com/sablier-labs/lockup/issues/1417.

### Checklist

- [x] Bob Integration tests
- [x] Vault Share Integration tests
- [ ] Adapter Integration tests

### CODE REVIEW

**Comment by smol-ninja** [comment]
@andreivladbrg I have made some non-functional changes to the source code as realized when writing tests.

File: bob/src/interfaces/IBobVaultShare.sol [hunk 1]
```diff
@@ -1,11 +1,12 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity >=0.8.22;
 
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
 
 /// @title IBobVaultShare
 /// @notice Interface for the ERC-20 token representing shares in a Bob vault.
-interface IBobVaultShare is IERC20Metadata {
+interface IBobVaultShare is IERC20, IERC20Metadata {
```
Comment (andreivladbrg) [inline bob/src/interfaces/IBobVaultShare.sol]: `IERC20Metadata` already inherits `IERC20`

File: bob/src/SablierBob.sol
Comment (andreivladbrg) [inline bob/src/SablierBob.sol]: do we still need this?

File: bob/tests/bob/Base.t.sol [hunk 2]
```diff
@@ -35,18 +41,18 @@ abstract contract Base_Test is Assertions, Modifiers {
...
     ISablierBob internal bob;
-    SablierLidoAdapter internal adapter;
-    MockAdapterInvalidInterface internal mockAdapterInvalid;
+    ISablierLidoAdapter internal adapter;
+    AggregatorV3Interface internal chainlinkOracle;
     ChainlinkOracleMock internal mockOracle;
     ChainlinkOracleWith18Decimals internal mockOracleInvalidDecimals;
     ChainlinkOracleZeroPrice internal mockOracleInvalidPrice;
     ChainlinkOracleWithRevertingDecimals internal mockOracleReverting;
     ChainlinkOracleWithRevertingPrice internal mockOracleRevertingOnLatestRoundData;
```
Comment (andreivladbrg) [inline bob/tests/bob/Base.t.sol]: we no longer need these reverting mocks

File: bob/tests/bob/Base.t.sol [hunk 3]
```diff
@@ -35,18 +41,18 @@ abstract contract Base_Test is Assertions, Modifiers {
     //////////////////////////////////////////////////////////////////////////*/
 
     ISablierBob internal bob;
-    SablierLidoAdapter internal adapter;
-    MockAdapterInvalidInterface internal mockAdapterInvalid;
+    ISablierLidoAdapter internal adapter;
+    AggregatorV3Interface internal chainlinkOracle;
     ChainlinkOracleMock internal mockOracle;
```
Comment (andreivladbrg) [inline bob/tests/bob/Base.t.sol]: we can use the oracle from the `"@sablier/evm-utils/src/tests/BaseTest.sol"`

File: bob/tests/bob/Base.t.sol [hunk 4]
```diff
@@ -83,24 +94,22 @@ abstract contract Base_Test is Assertions, Modifiers {
         // Deploy the real adapter with external mocks.
         deployAdapter();
```
Comment (andreivladbrg) [inline bob/tests/bob/Base.t.sol]: why isn't this in the `deployProtocol`?

File: bob/tests/bob/Base.t.sol [hunk 5]
```diff
@@ -35,18 +41,18 @@ abstract contract Base_Test is Assertions, Modifiers {
     //////////////////////////////////////////////////////////////////////////*/
 
     ISablierBob internal bob;
-    SablierLidoAdapter internal adapter;
-    MockAdapterInvalidInterface internal mockAdapterInvalid;
+    ISablierLidoAdapter internal adapter;
+    AggregatorV3Interface internal chainlinkOracle;
```
Comment (andreivladbrg) [inline bob/tests/bob/Base.t.sol]: why do we need this? we can import `AggregatorV3Interface` in mock contract

File: bob/tests/bob/Base.t.sol [hunk 6]
```diff
@@ -110,27 +119,26 @@ abstract contract Base_Test is Assertions, Modifiers {
...
         users.depositor = createUser("Depositor", spenders);
-        users.depositor2 = createUser("Depositor2", spenders);
+        users.bob = createUser("Bob", spenders);
 
-        // Give ETH to users and deposit into WETH (so WETH contract has ETH backing).
+        // Give ETH to users, deposit into WETH, and approve Bob.
         vm.deal(users.depositor, 1000 ether);
-        vm.deal(users.depositor2, 1000 ether);
+        vm.deal(users.bob, 1000 ether);
         vm.deal(users.alice, 1000 ether);
```
Comment (andreivladbrg) [inline bob/tests/bob/Base.t.sol]: we already use `vm.deal` in the `createUser` function

File: bob/tests/bob/Base.t.sol [hunk 7]
```diff
@@ -110,27 +119,26 @@ abstract contract Base_Test is Assertions, Modifiers {
...
+        IWETH9(address(weth)).deposit{ value: 1000 ether }();
+        weth.approve(address(bob), type(uint256).max);
         vm.stopPrank();
 
         vm.startPrank(users.alice);
-        weth.deposit{ value: 1000 ether }();
-        IERC20(address(weth)).approve(address(bob), type(uint256).max);
+        IWETH9(address(weth)).deposit{ value: 1000 ether }();
+        weth.approve(address(bob), type(uint256).max);
         vm.stopPrank();
```
Comment (andreivladbrg) [inline bob/tests/bob/Base.t.sol]: why not use `setMsgSender` instead of `vm.startPrank && vm.stopPrank`?

**Review by andreivladbrg — COMMENTED** [review]
leaving these comments for now

as discussed on slack, pushed this commit e71345838b87f064f032ae19568a71e765f67f24 which addresses majority of the comments

File: bob/tests/bob/utils/Constants.sol [hunk 8]
```diff
@@ -6,9 +6,9 @@ import { BaseConstants } from "@sablier/evm-utils/src/tests/BaseConstants.sol";
 
 abstract contract Constants is BaseConstants {
     // Amounts
-    uint128 public constant CURRENT_PRICE = 2000e18;
+    uint128 public constant CURRENT_PRICE = 3000e8;
```
Comment (smol-ninja) [inline bob/tests/bob/utils/Constants.sol:9]: Why does it matter if its 2000 or 3000? :))

File: utils/src/mocks/ChainlinkMocks.sol [hunk 9]
```diff
@@ -24,13 +24,28 @@ contract SafeOracleMock {
...
 /// @notice A mock Chainlink oracle that returns a $3000 price with 8 decimals.
-contract ChainlinkOracleMock {
+contract ChainlinkOracleMock is AggregatorV3Interface {
     int256 internal _price = 3000e8;
 
     function decimals() external pure returns (uint8) {
         return DEFAULT_DECIMALS;
     }
 
+    function description() external pure returns (string memory) {
```
Comment (smol-ninja) [inline utils/src/mocks/ChainlinkMocks.sol:34]: Why did you add this? Where are we using this function?

File: utils/src/mocks/ChainlinkMocks.sol [hunk 10]
```diff
@@ -47,6 +62,10 @@ contract ChainlinkOracleMock {
     function setPrice(uint256 newPrice) external {
         _price = int256(newPrice);
     }
+
+    function version() external pure returns (uint256) {
```
Comment (smol-ninja) [inline utils/src/mocks/ChainlinkMocks.sol:66]: Why did you add this? Where are we using this function?

Comment (andreivladbrg) [reply utils/src/mocks/ChainlinkMocks.sol:66]: its not used - now the mock imports the chainlink interface, and if we don't add all functions the contract requires to be `abstract`

File: bob/tests/bob/utils/Constants.sol (hunk 8 above)
Comment (andreivladbrg) [reply bob/tests/bob/utils/Constants.sol:9]: it does matter because the price from `comptroller.oracle()` in `@evm-utils` is set to `3000e8`, and the tests were failing because the min fee in wei didn’t equal the expected amount from the comptroller

Comment (andreivladbrg) [reply bob/tests/bob/utils/Constants.sol:9]: i think we can remove this and use the constant from `@evm-utils`

File: utils/src/mocks/ChainlinkMocks.sol (hunk 9 above)
Comment (andreivladbrg) [reply utils/src/mocks/ChainlinkMocks.sol:34]: same reason as [here](https://github.com/sablier-labs/lockup/pull/1420#discussion_r2852366445)

File: bob/tests/bob/utils/Constants.sol (hunk 8 above)
Comment (andreivladbrg) [reply bob/tests/bob/utils/Constants.sol:9]: just checked, it’s not worth it, as `ETH_PRICE_USD` is a `uint256` and we use `uint128` for the `targetPrice` in the bob protocol - so conversions would be required

File: bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree [hunk 11]
```diff
@@ -1,20 +1,32 @@
...
-   ├── when token is native token
+└── when token not zero
+   ├── when native token
    │  └── it should revert
-   └── when token not native token
-      ├── when expiry in past
+   └── when not native token
+      ├── when expiry not in future
+      │  └── it should revert
+      ├── when expiry in present
```
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree:10]: duplicated branch

this:

```tree
├── when expiry not in future
│  └── it should revert
├── when expiry in present
│  └── it should revert
└── when expiry in future
``` 


should become 

```tree
├── when expiry not in future
│  └── it should revert
└── when expiry in future
``` 

expiry in present covers the cases when the expiry is in the past

File: bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree [hunk 12]
```diff
@@ -1,20 +1,32 @@
...
-            ├── when target price at or below current price
+            ├── when target price not exceed oracle price
             │  └── it should revert
-            └── when target price above current price
-               ├── given no default adapter for token
-               │  └── it should create the vault without adapter
-               └── given default adapter for token
-                  └── it should create the vault with adapter
+            └── when target price exceeds oracle price
+               ├── given no adapter
```
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree:19]: just mentioning: missing the token symbol edge cases - tho already tested in the NFT descriptor from `lockup`

this would decrease the coverage

File: bob/tests/bob/integration/concrete/enter/enter.tree [hunk 13]
```diff
@@ -4,14 +4,25 @@ Enter_Integration_Concrete_Test
 └── given not null
    ├── given settled
    │  └── it should revert
-   └── given not settled
+   ├── given expired
+   │  └── it should revert
+   └── given active
```
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/enter/enter.tree]: statuses should be UPPER CASES

Comment (andreivladbrg) [reply bob/tests/bob/integration/concrete/enter/enter.tree]: also missing the case when the vault is _already_ settled or expired, not after syncing the price

File: bob/tests/bob/integration/concrete/enter/enter.tree [hunk 14]
```diff
@@ -4,14 +4,25 @@ Enter_Integration_Concrete_Test
...
       ├── when amount zero
       │  └── it should revert
       └── when amount not zero
          ├── given no adapter
-         │  ├── given first deposit
-         │  │  └── it should enter and set deposited at
-         │  └── given subsequent deposit
-         │     └── it should enter without updating deposited at
+         │  ├── when first deposit
```
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/enter/enter.tree]: why is this branch inside `given no adapter` ?

should them be:

```tree
├── when not first deposit
│  ├── it should not update the first deposit time
│  ├── it should transfer tokens to the contract
│  ├── it should mint share tokens to the caller
│  └── it should emit an {Enter} event
└── when first deposit
   ├── given no adapter
   │  ├── it should update the first deposit time
   │  ├── it should transfer tokens to the contract
   │  ├── it should mint share tokens to the caller
   │  └── it should emit an {Enter} event
   └── given adapter
      ├── it should update the first deposit time
      ├── it should transfer tokens to the adapter
      ├── it should stake via the adapter
      ├── it should mint share tokens to the caller
      └── it should emit an {Enter} and {Stake} events
```

File: bob/tests/bob/integration/concrete/lido-adapter/adapter.t.sol
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/lido-adapter/adapter.t.sol:1]: this test structure makes me think that the adapter is a function of the `SablierBob` contract and not a separate contract

why not move `SablierLidoAdapter` to a separate folder, like we have for the escrow contract, and test it separately? from its perspective, what matters is that the caller is the bob contract, which can be handled via `vm.prank`

File: bob/tests/bob/integration/Integration.t.sol [hunk 15]
```diff
@@ -1,79 +1,65 @@
...
-        mockOracle.setPrice(INITIAL_PRICE); // Reset for other tests.
     }
 
     /*//////////////////////////////////////////////////////////////////////////
                                 COMMON-REVERT-TESTS
     //////////////////////////////////////////////////////////////////////////*/
 
     /// @dev Expects a revert when the vault is null.
-    function expectRevert_Null(bytes memory callData, uint256 nullVaultId) internal {
+    function expectRevert_Null(bytes memory callData) internal {
```
Comment (andreivladbrg) [inline bob/tests/bob/integration/Integration.t.sol:71]: should we also add here the `expectRevert_SETTLED` and `expectRevert_EXPIRED` as it used in multiple functions?

File: bob/tests/bob/integration/concrete/lido-adapter/lido-adapter-constructor/constructor.tree [hunk 16]
```diff
@@ -0,0 +1,10 @@
+Constructor_LidoAdapter_Integration_Concrete_Test
+├── when slippage tolerance exceeds maximum
+│  └── it should revert
+├── when yield fee exceeds maximum
```
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/lido-adapter/lido-adapter-constructor/constructor.tree:4]: should we move the checks to a private function and not test these in the constructor?

`SablierLidoAdapter_SlippageToleranceTooHigh` and `SablierLidoAdapter_YieldFeeTooHigh`

File: bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.t.sol [hunk 17]
```diff
@@ -7,24 +7,29 @@ import { Errors as EvmUtilsErrors } from "@sablier/evm-utils/src/libraries/Error
 import { ISablierLidoAdapter } from "src/interfaces/ISablierLidoAdapter.sol";
 import { Errors } from "src/libraries/Errors.sol";
 
-import { Integration_Test } from "../../Integration.t.sol";
+import { Integration_Test } from "../../../Integration.t.sol";
 
 contract SetSlippageTolerance_Integration_Concrete_Test is Integration_Test {
+    function setUp() public override {
+        Integration_Test.setUp();
+        setMsgSender(users.bob);
```
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.t.sol]: lets remove the `setUp` function and add this `setMsgSender(users.bob);` in the `test_RevertWhen_CallerNotComptroller` function

File: bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.tree [hunk 18]
```diff
@@ -0,0 +1,15 @@
+SetSlippageTolerance_Integration_Concrete_Test
+├── when caller not comptroller
+│  └── it should revert
+└── when caller is comptroller
+   ├── when tolerance exceeds max
+   │  └── it should revert
+   ├── when tolerance within limit
```
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.tree:7]: missing TAB here?

File: bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.tree [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+   │  ├── it should update the slippage tolerance
+   │  └── it should emit a {SetSlippageTolerance} event
+   ├── when tolerance at maximum
```
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.tree:10]: do we need both (i) within limit and (ii) when tolerance at maximum branches?

File: bob/tests/bob/integration/concrete/lido-adapter/set-yield-fee/setYieldFee.t.sol [hunk 20]
```diff
@@ -7,28 +7,33 @@ import { Errors as EvmUtilsErrors } from "@sablier/evm-utils/src/libraries/Error
 import { ISablierBobAdapter } from "src/interfaces/ISablierBobAdapter.sol";
 import { Errors } from "src/libraries/Errors.sol";
 
-import { Integration_Test } from "../../Integration.t.sol";
+import { Integration_Test } from "../../../Integration.t.sol";
 
 contract SetYieldFee_Integration_Concrete_Test is Integration_Test {
+    function setUp() public override {
+        Integration_Test.setUp();
+        setMsgSender(users.bob);
```
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/lido-adapter/set-yield-fee/setYieldFee.t.sol]: similarly - we can remove the `setUp` function here and move the prank in the `test_RevertWhen_CallerNotComptroller` 

alternatively we can have common test (`expectRevert_CallerNotComptroller`) in the `Integration_Test` or add it in the `@evm-utils` as it would be useful in all protocols

File: bob/tests/bob/integration/concrete/lido-adapter/adapter.tree
Comment (andreivladbrg) [inline bob/tests/bob/integration/concrete/lido-adapter/adapter.tree:1]: why all functions in a single file test?

**Review by andreivladbrg — COMMENTED** [review]
leaving some more comments below

File: bob/src/SablierBob.sol
Comment (smol-ninja) [reply bob/src/SablierBob.sol]: Lets remove it. Though, we should implement it in the future version.

File: bob/tests/bob/Base.t.sol (hunk 2 above)
Comment (smol-ninja) [reply bob/tests/bob/Base.t.sol]: Yeah. I wanted to update Base in the end as I started reviewing tests using top down approach. I can see you have already reviewed it.

File: bob/tests/bob/utils/Constants.sol (hunk 8 above)
Comment (smol-ninja) [reply bob/tests/bob/utils/Constants.sol:9]: OK.

File: bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree (hunk 11 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree:10]: Good catch. I wanted the following:

```
├── when expiry in past
│  └── it should revert
├── when expiry in present
│  └── it should revert
└── when expiry in future
```

to cover all cases.

File: bob/tests/bob/integration/concrete/enter/enter.tree (hunk 13 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/enter/enter.tree]: > also missing the case when the vault is already settled or expired, not after syncing the price

If you mean the case where vault is settled or expired before syncing the price, then its covered. `given SETTLED` and `given EXPIRED` use `vaultIds.settledVault` and `vm.warp(EXPIRY + 1)` respectively.

> statuses should be UPPER CASES

https://github.com/sablier-labs/lockup/pull/1420/changes/b9374e2eaf0efebaea4ee9a988782e2bc8fae830

File: bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree (hunk 12 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree:19]: What if we move this logic into `utils` under a new library and then use it here as well as in NFT descriptor. Too overkill?

File: bob/tests/bob/integration/concrete/enter/enter.tree (hunk 14 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/enter/enter.tree]: We are going to remove grace period feature we will also remove first deposit checks.

File: bob/tests/bob/integration/concrete/lido-adapter/adapter.t.sol
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/lido-adapter/adapter.t.sol:1]: I have not updated Adapter and vault share tests, will do via separate PR.

Although the reason I did not create a separate folder for adapter similar to escrow is because its more or less related to the bob contract. This is similar to the approach we have in our Lockup testing where batch-lockup is placed in the same dir as lockup contract.

Though let me take a look at it when I review adapter tests.

File: bob/tests/bob/integration/concrete/lido-adapter/lido-adapter-constructor/constructor.tree (hunk 16 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/lido-adapter/lido-adapter-constructor/constructor.tree:4]: Yeah I will. As the PR checklist mentions, adapter and vault shares need to be reviewed by me.

File: bob/tests/bob/integration/Integration.t.sol (hunk 15 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/Integration.t.sol:71]: Good idea will do so.

File: bob/src/SablierBob.sol
Comment (smol-ninja) [reply bob/src/SablierBob.sol]: https://github.com/sablier-labs/lockup/pull/1420/changes/b9374e2eaf0efebaea4ee9a988782e2bc8fae830

File: bob/tests/bob/Base.t.sol (hunk 7 above)
Comment (smol-ninja) [reply bob/tests/bob/Base.t.sol]: https://github.com/sablier-labs/lockup/pull/1420/changes/b9374e2eaf0efebaea4ee9a988782e2bc8fae830

File: bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree (hunk 11 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree:10]: https://github.com/sablier-labs/lockup/pull/1420/changes/b9374e2eaf0efebaea4ee9a988782e2bc8fae830

File: bob/tests/bob/integration/Integration.t.sol (hunk 15 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/Integration.t.sol:71]: https://github.com/sablier-labs/lockup/pull/1420/changes/b9374e2eaf0efebaea4ee9a988782e2bc8fae830

File: bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree (hunk 12 above)
Comment (andreivladbrg) [reply bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree:19]: > What if we move this logic into `utils` under a new library and then use it here as well as in NFT descriptor. Too overkill?

we can do this, yeah

but i'd say if we end up allocating too much time to this, we should give up on the idea

File: bob/tests/bob/integration/concrete/enter/enter.tree (hunk 13 above)
Comment (andreivladbrg) [reply bob/tests/bob/integration/concrete/enter/enter.tree]: > If you mean the case where vault is settled or expired before syncing the price, then its covered. `given SETTLED` and `given EXPIRED` use `vaultIds.settledVault` and `vm.warp(EXPIRY + 1)` respectively.
> 
> >

i am talking the case when after calling `_syncPriceFromOracle`  changes the status

similar to redeem tests

https://github.com/sablier-labs/lockup/blob/b9374e2eaf0efebaea4ee9a988782e2bc8fae830/bob/tests/bob/integration/concrete/redeem/redeem.tree#L5-L18

File: bob/tests/bob/integration/concrete/enter/enter.tree (hunk 14 above)
Comment (andreivladbrg) [reply bob/tests/bob/integration/concrete/enter/enter.tree]: right -lets keep it like this and we refactor later

File: bob/tests/bob/integration/concrete/lido-adapter/adapter.t.sol
Comment (andreivladbrg) [reply bob/tests/bob/integration/concrete/lido-adapter/adapter.t.sol:1]: > This is similar to the approach we have in our Lockup testing where batch-lockup is placed in the same dir as lockup contract

if we want to have the structure, we'd need to add a new dir in the concrete folder called bob and move the functions to have only contracts there:

### lockup

image

### bob

image

File: bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree (hunk 12 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree:19]: We can ask Claude to do it. It should be one shot for t. I will try.

File: bob/tests/bob/integration/concrete/lido-adapter/adapter.t.sol
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/lido-adapter/adapter.t.sol:1]: OK will do

File: bob/tests/bob/integration/concrete/enter/enter.tree (hunk 13 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/enter/enter.tree]: https://github.com/sablier-labs/lockup/pull/1420/changes/3b1ad771ff0be82c1c6430a859dc4034ecaa5b73

Looks good now?

File: bob/tests/bob/integration/concrete/lido-adapter/adapter.t.sol
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/lido-adapter/adapter.t.sol:1]: https://github.com/sablier-labs/lockup/pull/1420/changes/2afcdc3c820630d8343950630138ae9049806273

**Comment by smol-ninja** [comment]
@andreivladbrg if it looks good now, can we merge it please? I will open another PR fixing ~vault share and~ adapter tests (I already have them locally just reviewing a bunch more). This way, we can parallelize the audit fixes or you can add fork/fuzz for Bob through a PR.

File: bob/tests/bob/integration/concrete/bob/enter/enter.t.sol [hunk 21]
```diff
@@ -4,108 +4,160 @@ pragma solidity >=0.8.22 <0.9.0;
...
 import { Integration_Test } from "../../Integration.t.sol";
 
 contract Enter_Integration_Concrete_Test is Integration_Test {
+    /// @dev The share token for the default vault.
+    IERC20 internal shareToken;
+
+    function setUp() public override {
+        Integration_Test.setUp();
+
+        // Change called to bob for this test.
```
[the remark this answers is not in the document]
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/bob/enter/enter.t.sol]: https://github.com/sablier-labs/lockup/pull/1420/commits/959b3152dc9f954d7567116240caaee32743ef62

File: bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree (hunk 12 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/bob/create-vault/createVault.tree:19]: https://github.com/sablier-labs/lockup/pull/1424

File: bob/tests/bob/integration/concrete/lido-adapter/lido-adapter-constructor/constructor.tree (hunk 16 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/lido-adapter/lido-adapter-constructor/constructor.tree:4]: Why did you want to move these to a private function and not test these in the constructor? They are related to constructor so I think it makes sense to have them in the constructor with a proper BTT tree.

File: bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.t.sol (hunk 17 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.t.sol]: https://github.com/sablier-labs/lockup/pull/1425

File: bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.tree (hunk 18 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.tree:7]: https://github.com/sablier-labs/lockup/pull/1425

File: bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.tree (hunk 19 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/lido-adapter/set-slippage-tolerance/setSlippageTolerance.tree:10]: I hope it is resolved in https://github.com/sablier-labs/lockup/pull/1425

```tree
SetSlippageTolerance_Integration_Concrete_Test
├── when caller not comptroller
│  └── it should revert
└── when caller comptroller
   ├── when slippage tolerance exceeds max
   │  └── it should revert
   └── when slippage tolerance not exceed max
      ├── it should update the new slippage tolerance
      └── it should emit a {SetSlippageTolerance} event
```

File: bob/tests/bob/integration/concrete/lido-adapter/set-yield-fee/setYieldFee.t.sol (hunk 20 above)
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/lido-adapter/set-yield-fee/setYieldFee.t.sol]: https://github.com/sablier-labs/lockup/pull/1425

File: bob/tests/bob/integration/concrete/lido-adapter/adapter.tree
Comment (smol-ninja) [reply bob/tests/bob/integration/concrete/lido-adapter/adapter.tree:1]: fixed in https://github.com/sablier-labs/lockup/pull/1425
